### PR TITLE
Structure Nomios web app

### PR DESCRIPTION
**Pull request by satazor**
See https://github.com/ipfs-shipyard/pm-idm/issues/104

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,7 +1,6 @@
 {
   "name": "nomios-web",
   "description": "Codebase for the web version of Nomios, a reference implementation of the IDM Wallet UI",
-  "homepage": "https://github.com/ipfs-shipyard/nomios-web#readme",
```
Comment (PedroMiguelSS) [inline package.json:4]: Why did you remove this line?

File: src/modals/lock-screen/LockScreen.js [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+        this.setState({ promise });
+    }
+
+    handleInputChange = (event) => {
+        const value = event.target.value;
+
+        this.setState(({ promise }) => ({
+            // Reset passphrase if there was error
+            passphrase: getPromiseState(promise).status === 'rejected' ? value.substr(-1, 1) : value,
```
Comment (dominguesgm) [inline src/modals/lock-screen/LockScreen.js]: According to MDN, `substr()` is a legacy function. Maybe we could replace it with `value.substring(value.length-1)`.

File: src/modals/lock-screen/LockScreen.js [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+                    {
+                        Array(passphraseLength).fill(0)
+                        .map((val, i) => (
+                            <CSSTransition
+                                classNames={ passphraseDotsClassNames }
+                                key={ passphraseLength - i }
+                                timeout={ 50 }
+                                exit={ status !== 'rejected' }>
+                                <div
+                                    className={ classNames(styles.passphraseDot, status === 'pending' && styles.loading) }
```
Comment (dominguesgm) [inline src/modals/lock-screen/LockScreen.js]: The dots should animate even if `status` is fulfilled, otherwise the loading dots animation won't have any time to run.

File: src/modals/lock-screen/LockScreen.js [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+        const value = event.target.value;
+
+        this.setState(({ promise }) => ({
+            // Reset passphrase if there was error
+            passphrase: getPromiseState(promise).status === 'rejected' ? value.substr(-1, 1) : value,
+            promise: undefined,
+        }));
+    };
+
+    handleInputKeyDown = (event) => {
```
Comment (dominguesgm) [inline src/modals/lock-screen/LockScreen.js:186]: I'm having a strange behavior when testing: if I press a key immediately after pressing enter, it is still inserted in the input. This, however, was not happening in the previous version.

This is probably caused by the introduction of `PromiseState`. I will spend some time looking into this issue.

**Review by dominguesgm — CHANGES_REQUESTED** [review]

**Review by dominguesgm — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (PedroMiguelSS) [reply package.json:4]: I got it. CRA uses this URL on the build process.
